Source/Core/Core/Debugger/PPCDebugInterface.cpp

```diff
@@ -324,6 +325,40 @@ std::string PPCDebugInterface::GetDescription(unsigned int address)
   return g_symbolDB.GetDescription(address);
 }
 
+u32 PPCDebugInterface::GetMemoryAddressFromInstruction(std::string instruction)
```

Comment:
It may be preferable to use a `std::optional<u32>` for this. To handle the mentioned error case relating to checking the return value of `std::regex_search`.

Reply:
This is a tough one. For Trace I always need this to reliably return a memory address. I filter what can feed into this, to make sure it's a load/store, and so it should reliably work. If I could think of an instance of it failing, I could decide on how handle it. I'm thinking maybe just return 0 for false.